pkg/util/test/util.go

```diff
@@ -427,26 +430,41 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
 	if len(kubeconfig) != 0 && kubeconfig[0] != "" {
 		os.Setenv("KUBECONFIG", kubeconfig[0])
 	}
-	cluster, err = operatorops.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
-	if err != nil && !errors.IsNotFound(err) {
-		return err
-	}
-	if cluster.Spec.DeleteStrategy == nil ||
-		(cluster.Spec.DeleteStrategy.Type != corev1.UninstallAndWipeStorageClusterStrategyType &&
-			cluster.Spec.DeleteStrategy.Type != corev1.UninstallStorageClusterStrategyType) {
-		cluster.Spec.DeleteStrategy = &corev1.StorageClusterDeleteStrategy{
-			Type: corev1.UninstallAndWipeStorageClusterStrategyType,
+
+	t := func() (interface{}, bool, error) {
+		cluster, err = operatorops.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+		if err != nil && !errors.IsNotFound(err) {
+			return nil, true, err
 		}
-		if _, err := operatorops.Instance().UpdateStorageCluster(cluster); err != nil {
-			return err
+
+		if cluster.Spec.DeleteStrategy == nil ||
+			(cluster.Spec.DeleteStrategy.Type != corev1.UninstallAndWipeStorageClusterStrategyType &&
+				cluster.Spec.DeleteStrategy.Type != corev1.UninstallStorageClusterStrategyType) {
+			cluster.Spec.DeleteStrategy = &corev1.StorageClusterDeleteStrategy{
+				Type: corev1.UninstallAndWipeStorageClusterStrategyType,
+			}
+
+			if _, err := operatorops.Instance().UpdateStorageCluster(cluster); err != nil {
```

Comment:
nit: it might be better to get the latest cluster here for the validation and deletion later? to avoid running into object version issues. I don't know if that would happen, so probably it's ok

Reply:
We do not need to get new `StorageCluster` object as `operatorops.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace)` only cares about name and namespace